modules/RsLedDriver/src/main.rs

```diff
@@ -0,0 +1,135 @@
+#![allow(non_snake_case)]
+include!(concat!(env!("OUT_DIR"), "/generated.rs"));
+
+use generated::{get_config, Context, Module, ModulePublisher};
+use generated::{
+    EmptyServiceSubscriber, EvseManagerClientSubscriber, LedDriverClientPublisher,
+    LedDriverClientSubscriber, OnReadySubscriber,
+};
+
+use generated::types::energy::EnforcedLimits;
+use generated::types::evse_board_support::{HardwareCapabilities, Telemetry};
+use generated::types::evse_manager::*;
+use generated::types::iso15118_charger::RequestExiStreamSchema;
+use generated::types::led_state::LedState;
+use generated::types::powermeter::Powermeter;
+
+use std::sync::{Arc, Mutex};
+use std::{thread::sleep, time::Duration};
+
+fn get_led_state(trigger: SessionEventEnum) -> Option<LedState> {
+    match trigger {
+        SessionEventEnum::Disabled
+        | SessionEventEnum::AuthRequired
+        | SessionEventEnum::PluginTimeout => Some(LedState::Red),
+
+        // Blue triggers
+        SessionEventEnum::TransactionStarted
+        | SessionEventEnum::PrepareCharging
+        | SessionEventEnum::ChargingStarted
+        | SessionEventEnum::ChargingPausedEV
+        | SessionEventEnum::ChargingPausedEVSE
+        | SessionEventEnum::WaitingForEnergy
+        | SessionEventEnum::ChargingResumed
+        | SessionEventEnum::StoppingCharging
+        | SessionEventEnum::ChargingFinished
+        | SessionEventEnum::TransactionFinished
+        | SessionEventEnum::SessionResumed => Some(LedState::Blue),
+
+        // Green triggers
+        SessionEventEnum::Authorized
+        | SessionEventEnum::Deauthorized
+        | SessionEventEnum::Enabled
+        | SessionEventEnum::SessionFinished
+        | SessionEventEnum::ReservationStart
+        | SessionEventEnum::ReservationEnd => Some(LedState::Green),
+        _ => None,
+    }
+}
+
+pub struct LedDriver {
+    // This vector contains all the led_drivers.
+    // Every entry here matches the index of evse_manager in config
+    led_drivers: Mutex<Vec<LedDriverClientPublisher>>,
```

Comment:
you don't need this field. the context provides you the list

Reply:
Context provides me a list of publishers I can use? I think that context provide me the context of the EvseManagers and which one sends the SessionEvent

Reply:
yes it should be `context.publisher.led_driver_slots`